### DISCUSSION TITLE
Is it possible to use a huggingface Automodel(device_map = auto) with a multi-gpu trainer?

### DISCUSSION THREAD

**Discussion by surya-narayanan** [Lightning-AI/pytorch-lightning#19194] [DDP / multi-GPU / multi-node]
The automodel allocates the model across devices, but the L trainer seems to want to replicate model architecture across devices, which is incompatible. 

When I try using an automodel with device_map set to auto in a trainer, the forward pass errors out, claiming that it is multiplying tensors on different devices.

**Comment by noviceninja88** [comment]
@surya-narayanan hey, did you find any solution ?
